backend-api/template.yaml

```diff
@@ -216,13 +294,134 @@ Resources:
                 Action:
                   - secretsmanager:GetSecretValue
                 Resource:
-                  - !Sub "arn:aws:secretsmanager$:{AWS::Region}:${AWS::AccountId}:secret:SECRET_NAME-12345"
-                  
-        - PolicyName: AsyncTokenFunctionKmsPolicy
-          PolicyDocument:
-            Version: '2012-10-17'
-            Statement:
-              - Effect: Allow
-                Action:
-                  - kms:Sign
-                Resource: !ImportValue platform-kms-signing-key-arn
\ No newline at end of file
+                  - !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${Environment}/clientRegistry*"
```

Comment:
Why the wildcard at the end? Are we expecting multiple secrets with the same prefix?

Reply:
AWS adds a random 6 character string to the end of secrets - do you know if there's a way round this?

Reply:
Ah I follow now. 

I checked their docs for guidance and it is the same process, they just has a slightly more constrained approach: 
https://docs.aws.amazon.com/secretsmanager/latest/userguide/auth-and-access_examples.html#auth-and-access_examples_wildcard

![image](https://github.com/user-attachments/assets/f4aa939d-be22-4d59-923a-ab0b3fed4d00)

So this best we could get is: 

```suggestion
                  - !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${Environment}/clientRegistry-??????"
```

Since AWS always adds those 6 characters this blocks anyone sneaking a different secret in.

Reply:
done!